ansible_base/authentication/authenticator_plugins/oidc.py

```diff
@@ -38,6 +42,140 @@ class OpenIdConnectConfiguration(BaseAuthenticatorConfiguration):
         ui_field_label=_('OIDC Secret'),
     )
 
+    #################################
+    # Additional params
+    #################################
+
+    ACCESS_TOKEN_URL = URLField(
+        help_text=_("The URL to obtain an access token from the OIDC provider."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Access Token URL"),
+    )
+
+    ACCESS_TOKEN_METHOD = CharField(
+        help_text=_("The HTTP method to be used when requesting an access token. Typically 'POST' or 'GET'."),
+        default="POST",
+        allow_null=True,
+        required=False,
+        ui_field_label=_("Access Token Method"),
+    )
+
+    AUTHORIZATION_URL = URLField(
+        help_text=_("The URL to redirect the user for OIDC provider authorization."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Authorization URL"),
+    )
+
+    ID_KEY = CharField(
+        help_text=_("The JSON key used to extract the user's ID from the ID token."),
+        default="sub",
+        allow_null=True,
+        required=False,
+        ui_field_label=_("ID Key"),
+    )
+
+    ID_TOKEN_ISSUER = CharField(
+        help_text=_("Expected issuer ('iss') of the ID token. If set, it will be used to validate the issuer of the ID token."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("ID Token Issuer"),
+    )
+
+    ID_TOKEN_MAX_AGE = IntegerField(
+        help_text=_("The maximum allowed age (in seconds) of the ID token. Tokens older than this will be rejected."),
+        default=600,
+        allow_null=True,
+        ui_field_label=_('OIDC Token Max Age'),
+    )
+
+    JWT_ALGORITHMS = ListField(
```

Comment:
Can we extract the possible algorithms from the library and either add a custom validator (or can we add 'choices' to a ListField)?

Reply:
Apply this comment to any of the subsequent fields, anywhere we can do validation/choices is 👍